lib/Tools/circt-bmc/LowerToBMC.cpp

```diff
@@ -60,6 +61,8 @@ void LowerToBMCPass::runOnOperation() {
     return signalPassFailure();
   }
 
+  sortTopologically(&hwModule.getBodyRegion().front());
```

Comment:
nit: Can you check if they are topologically sorted? 
```suggestion
  if(!sortTopologically(&hwModule.getBodyRegion().front())) {
    moduleOp.emitError("...");
    return signalPassFailure();
   }
```

Reply:
good point, thanks!